Sources/Core/Renderer/RenderCoordinator.swift

```diff
@@ -3,6 +3,48 @@ import FirebladeMath
 import MetalKit
 import DeltaCore
 
+public struct RendererError: LocalizedError {
+  public enum ErrorKind {
+    case getMetalDevice
+    case makeRenderCommandQueue
+    case camera
+    case skyBoxRenderer
+    case worldRenderer
+    case guiRenderer
+    case screenRenderer
+    case depthState
+  }
+  
+  public let kind: ErrorKind
+  public private(set) var error: Error? = nil
+  
+  public var errorDescription: String? {
+    switch kind {
+      case .getMetalDevice:
+        return "Failed to get metal device"
+      case .makeRenderCommandQueue:
+        return "Failed to make render command queue"
+      case.camera:
+        return "Failed to create camera: \(String(describing: error!)) - \(error!.localizedDescription)"
```

Comment:
this is for stackotter's discretion but generally i've been told before to avoid force unwrapping in my contributions

Reply:
i'll let stackotter decide on this one. it should be safe to force unwrap here as in those cases an error is always passed

Reply:
You might also be able to get around this (and the empty `catch {}`) with enum associated values. I’ll suggest a change and let me know if it works.

Reply:
I made a PR on your repo. If you like the format feel free to use it. If you’d like to stay with Optionals for simplicity, that works too.

Reply:
This has been improved (and solved) thanks to @obvgab's commit